tm2/pkg/crypto/keys/client/common.go

```diff
@@ -20,21 +22,34 @@ var DefaultBaseOptions = BaseOptions{
 }
 
 func HomeDir() string {
-	// if environment set, always use that.
-	// if not, check whether can get os.UserHomeDir()
-	// if not, fall back to home directory
+	// if environment variable is set, always use that.
+	// otherwise, use config dir (varies depending on OS) + "gno"
 	var err error
 	dir := os.Getenv("GNO_HOME")
 	if dir != "" {
 		return dir
 	}
 	dir, err = os.UserConfigDir()
-	if err == nil {
-		return filepath.Join(dir, "gno")
+	if err != nil {
+		panic(fmt.Errorf("couldn't get user config dir: %w", err))
 	}
-	dir, err = os.UserHomeDir()
+	gnoHome := filepath.Join(dir, "gno")
+	// XXX: added april 2023 as a transitory measure - remove after test4
+	fixOldDefaultGnoHome(gnoHome)
+	return gnoHome
+}
+
+func fixOldDefaultGnoHome(newDir string) {
+	dir, err := os.UserHomeDir()
 	if err != nil {
-		panic(err)
+		return
+	}
+	oldDir := filepath.Join(dir, ".gno")
+	s, err := os.Stat(oldDir)
+	if err != nil || !s.IsDir() {
+		return
+	}
+	if err = os.Rename(oldDir, newDir); err != nil && !os.IsExist(err) {
```

Comment:
Just pushed a commit to give a warning in that case. We get a "file exists" error (ie os.IsExist == true) -- this will generally be the case where a user has used gno commands before this change, but after #719